src/components/PointsSidePanels.js

```diff
@@ -0,0 +1,128 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import Typography from '@mui/material/Typography';
+import {
+  ADD_RED_MODE,
+  REMOVE_RED_MODE,
+  ADD_BLUE_MODE,
+  REMOVE_BLUE_MODE,
+} from '../config';
+import Tooltip from '@mui/material/Tooltip';
+import Button from '@mui/material/Button';
+import DeletePointsModal from './DeletePointsModal';
+//GEOCOMPONENTS
+import ButtonGroupList from '@geomatico/geocomponents/ButtonGroupList';
+// UTILS
+import { useTranslation } from 'react-i18next';
+import AddIcon from '@mui/icons-material/AddLocationAlt';
+import RemoveIcon from '@mui/icons-material/WrongLocation';
+
+function PointsSidePanels({onChangePoints, onChangeModePoints }) {
+  const { t } = useTranslation();
+ 
+  const [selectedMode, setSelectedMode] = useState('bluePoints-ADD_BLUE');
+  const handleFlatClick = (newMode) => {
+    if (newMode) {
+      setSelectedMode(newMode);
+      const changeMode = newMode.slice(newMode.indexOf('-') + 1);
+      onChangeModePoints(changeMode);
+    }
+  };
+  const [openModal, setOpenModal] = useState(false);
+  const handleOpenModal = () => {
+    setOpenModal(true);
+  };
+  const buttonGroupItemsBlue = [
+    {
+      id: ADD_BLUE_MODE,
+      content: 
+        <Tooltip title={t('add_poi')}>
+          <AddIcon />
+        </Tooltip>
+      
+    },
+    {
+      id: REMOVE_BLUE_MODE,
+      content: 
+        <Tooltip title={t('remove_poi')}>
+          <RemoveIcon />
+        </Tooltip>
+      
+    },
+  ];
+
+  const buttonGroupItemsRed = [
+    {
+      id: ADD_RED_MODE,
+      content: 
+        <Tooltip title={t('add_origin')}>
+          <AddIcon />
+        </Tooltip>
+      
+    },
+    {
+      id: REMOVE_RED_MODE,
+      content: 
+        <Tooltip title={t('remove_origin')}>
+          <RemoveIcon />
+        </Tooltip>
+      
+    },
+  ];
+
+  const categoriesBlue = [
+    {
+      id: 'bluePoints',
+      description: <Typography>{t('originPoints')}</Typography>,
+    },
+  ];
+
+  const categoriesRed = [
+    {
+      id: 'redPoints',
+      description: <Typography>{t('finalPoints')}</Typography>,
+    },
+  ];
+
+  return (
+    <>
+      <ButtonGroupList
+        buttonGroupVariant='outlined'
+        categories={categoriesBlue}
+        buttonGroupItems={buttonGroupItemsBlue}
+        selectedItemId={selectedMode}
+        onItemClick={handleFlatClick}
+      />
+
+      <ButtonGroupList
+        buttonGroupVariant='outlined'
+        buttonGroupColor='#d70f0f'
+        categories={categoriesRed}
+        buttonGroupItems={buttonGroupItemsRed}
+        selectedItemId={selectedMode}
+        onItemClick={handleFlatClick}
+      />
```

Comment:
La gracia del buttonGroupList es que tienes tanto las opciones azules como las rojas en un único componente y el se encarga de que si haces click en añadir azul (por ejemplo) se desmarca lo que esté seleccionado de las otras opciones, a los efectos hay cuatro posibilidades pero solo una clickada, como es nuestro ejemplo.

No se si se me entiende pero solo necesitas un buttonGroup, aunque reconozco que igual hay que hacer malabares para que los dos primeros sean rojos y los dos últimos azules

Reply:
Por si te sirve de algo puedes consultar un ejemplo:
[https://github.com/geomatico/irriter/blob/709d451fd740c1ed70333626e433eda06c677f7c/client/src/components/VariablePicker.tsx#L6](url)
Y el propio componente en storybook:
[https://labs.geomatico.es/geocomponents/?path=/story/forms-buttongrouplist--default](url)
En storybook activa la pestaña de Actions y mira lo que devuelve `onItemClick`
![image](https://github.com/geomatico/the-perfect-spot/assets/25547057/becd7467-83ef-4d57-8ecf-f74202698892)